pkg/virt-api/webhooks/mutating-webhook/mutators/vmi-mutator_test.go

```diff
@@ -414,42 +415,45 @@ var _ = Describe("VirtualMachineInstance Mutator", func() {
 		),
 	)
 
-	DescribeTable("should add the default network interface",
```

Comment:
```
DescribeTable("should add the default network interface",
		func(iface string, field gstruct.Field) {
			permit := true
			testutils.UpdateFakeKubeVirtClusterConfig(kvInformer, &v1.KubeVirt{
				Spec: v1.KubeVirtSpec{
					Configuration: v1.KubeVirtConfiguration{
						NetworkConfiguration: &v1.NetworkConfiguration{
							NetworkInterface:     expectedIface,
							PermitSlirpInterface: &permit,
						},
					},
				},
			})

			_, vmiSpec, _ := getMetaSpecStatusFromAdmit(rt.GOARCH)
                       Expect(vmiSpec.Domain.Devices.Interfaces[0]).To(gstruct.MatchFields(gstruct.IgnoreExtras,
					gstruct.Fields{
						"InterfaceBindingMethod": gstruct.MatchFields(gstruct.IgnoreExtras, field),
					},
				))
			
		},
		Entry("as bridge", "bridge", gstruct.Fields{"Bridge": Not(BeNil())}),
		Entry("as masquerade", "masquerade", , gstruct.Fields{"Masquerade": Not(BeNil())}),
		Entry("as slirp", "slirp" , gstruct.Fields{"Slirp": Not(BeNil())}),
	)
```

Reply:
This is interesting.
But it is IMO hard to read.

I'm supposed to be able and understand the test quickly, but here the cognitive effort needed for me to interpret this is high.

Reply:
I agree with @EdDev on this. 
This is cool, but much less readable IMO. Also, with your implementation we permit slirp interface also in other interfaces tests, which IIUC is not needed and is only confusing.

Reply:
> This is interesting. But it is IMO hard to read.
> 
> I'm supposed to be able and understand the test quickly, but here the cognitive effort needed for me to interpret this is high.

What is the cognitive load here?

Reply:
I am still not sure what is hard about this... I think it reads pretty well, match fields, field InterfaceBindingMethod needs to match fields based on parameter...

Reply:
I just realized, that there is a straightforward option here that keeps using the table.
@assafad , as the PR got an automatic hold due to too many retries, perhaps it is a good opportunity to consider it (if it really works).

```go
DescribeTable("should add the default network interface", func(defaultBindingMethod string, expectedIfaceBindingMethod v1.InterfaceBindingMethod) {
	testutils.UpdateFakeKubeVirtClusterConfig(kvInformer, &v1.KubeVirt{
		Spec: v1.KubeVirtSpec{
			Configuration: v1.KubeVirtConfiguration{
				NetworkConfiguration: &v1.NetworkConfiguration{
					NetworkInterface:     expectedIface,
					PermitSlirpInterface: pointer.P(true),
				},
			},
		},
	})
	_, vmiSpec, _ := getMetaSpecStatusFromAdmit(rt.GOARCH)
	Expect(vmiSpec.Domain.Devices.Interfaces[0].InterfaceBindingMethod).To(Equal(expectedIfaceBindingMethod))
},
    Entry("as bridge", "bridge", v1.InterfaceBindingMethod{Bridge: &v1.InterfaceBridge{}}),
    Entry("as masqurade", "masquerade", v1.InterfaceBindingMethod{Masquerade: &v1.InterfaceMasquerade{}}),
    Entry("as slirp", "slirp", v1.InterfaceBindingMethod{Slirp: &v1.InterfaceSlirp{}}),
)
```

Reply:
Thanks, it works. Done.